### PR TITLE
[JENKINS-49861] Create a non root user for running Jenkins and the evergreen-client

### PR DESCRIPTION
**Pull request by batmat**
Cf. [JENKINS-50195](https://issues.jenkins-ci.org/browse/JENKINS-50195)

### CODE REVIEW

**Comment by batmat** [comment]
OK, *now* test is failing for the reason I was expecting. Pushing the fixing code now.
```
INFO: Jenkins is fully up and running
Jenkins has started.
test_smoke
test_docker_CLI_available
test_no_executor
test_not_root
ASSERT:expected:<jenkins> but was:<root>
ASSERT:expected:<jenkins> but was:<root>
ASSERT:expected:<jenkins> but was:<root>
ASSERT:expected:<jenkins> but was:<root>
ASSERT:expected:<jenkins> but was:<root>

Cleaning up... G'day!

Ran 4 tests
```

File: Dockerfile [hunk 1]
```diff
@@ -85,7 +81,16 @@ COPY build/configuration-as-code/target/configuration-as-code.hpi /usr/share/jen
...
+# can be persisted and survive image upgrades
+VOLUME ${EVERGREEN_HOME}
+
 # Ensure the supervisord configuration is copied and executed by default such
 # that the Jenkins and evergreen-client processes both execute properly
 COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf
 CMD /usr/bin/supervisord -c /etc/supervisor/conf.d/supervisord.conf
+
+WORKDIR $EVERGREEN_HOME
+USER jenkins
```
Comment (rtyler) [inline Dockerfile]: I thought this should be `$user` rather than the constant string `jenkins`?

Comment (batmat) [reply Dockerfile]: Absolutely, good catch. I had that locally at some point when debugging the chown/volume issue, and forgot to re-adjust. Fixing

**Comment by batmat** [comment]
\o/

**Pull request merged**